### ISSUE TITLE
Firing Multiple Tags at Once

### ISSUE THREAD

**Issue by c0** [rentpath/jquery-autotagging.js#8]
Akil and I discovered that firing tags is essentially synchronous and only one can be fired one at a time. This came out of work on https://github.com/primedia/ag/pull/1352 when we needed to fire a room planner timer and a the tab change event.

One solution is to stop reusing the tag defined in https://github.com/primedia/jquery-autotagging/blob/dev/src/jquery.autotagging.coffee#L160 Instead, create new a new img element for each fire.

Our workaround was to delay the firing for 100ms.

**Comment by kaborso** [comment]
[The current approach](https://github.com/primedia/jquery-autotagging/blob/dev/src/jquery.autotagging.coffee#L160) (also below) assumes that the tagging pixel will load quickly and simply changes the `src` of the `<img/>` element (after creating it and storing it in `@warehouseTag`).

``` coffee
unless @warehouseTag
  @warehouseTag = $('<img/>',
    {id:'PRMWarehouseTag', border:'0', width:'1', height:'1' })

@warehouseTag.onload = $('body').trigger('WH_pixel_success_' + obj.type)
@warehouseTag.onerror = $('body').trigger('WH_pixel_error_' + obj.type)

@warehouseTag[0].src = requestURL
```

A possible solution is to create an array and push local `warehouseTag` variables to it so that we no longer have just one instance level `warehouseTag` which has to change src and onload/onerror callbacks per pixel fire. Each element will have its own handlers and src.

``` coffee
unless @warehouseTags
  @warehouseTags = []

warehouseTag = $('<img/>',
 {id:'PRMWarehouseTag', border:'0', width:'1', height:'1' })

warehouseTag.onload = $('body').trigger('WH_pixel_success_' + obj.type)
warehouseTag.onerror = $('body').trigger('WH_pixel_error_' + obj.type)

warehouseTag[0].src = requestURL
@warehouseTags.push warehouseTag
```

**Comment by kaborso** [comment]
Err, I guess we could simply use a local warehouseTag variable without the array.

**Comment by c0** [comment]
@martinstreicher can you weigh in on this?

**Comment by mpelzsherman** [comment]
Can you please summarize what is prevent the events from firing asynchronously?

Is this issue specific to image elements, or does it impact every WH tag?

In general, I do not like workarounds that involve arbitrary delays.

**Comment by c0** [comment]
`@warehouseTag` is reused and assigned `src` two times in a row. The second fire changes the src so fast that the first one doesn't give the browser enough time to request the src.

@mpelzsherman I agree with you on arbitrary delays, that's why we opened the issue to discuss.

**Comment by kaborso** [comment]
@mpelzsherman a WH tag is actually an `<img>` element. In fact, it is a single `<img>` element which is used over and over again. If multiple tags are fired rapidly, the `src` (all the tag info is on the query string) will be overwritten for each one thus canceling requests before they can reach the warehouse.

**Comment by c0** [comment]
#26 does not reuse the image tag.

**Comment by c0** [comment]
Fixed by #26
